### PR TITLE
Allow decorating a class to indicate decorating all methods defined w…

### LINKED ISSUE #2: Allow decorating a class to indicate decorating all methods defined within
The goal is that this:

```python
from birdseye import eye


def baz(self):
    print(3)


@eye
class A(object):
    b = baz

    def foo(self):
        pass

    def bar(self):
        pass
```

becomes equivalent to:

```python
@eye
def baz(self):
    print(3)


class A(object):
    b = baz

    @eye
    def foo(self):
        pass

    @eye
    def bar(self):
        pass
```

Some handy code as a starting point:

    [x for x in A.__dict__.values() if inspect.ismethod(x) or inspect.isfunction(x)]

### PR DESCRIPTION
**Pull request by bugosiphionah**
Closes #2

### CODE REVIEW

File: birdseye/__init__.py [hunk 1]
```diff
@@ -292,7 +298,6 @@ def get_start(n):
 def _deep_dict():
     return defaultdict(_deep_dict)
 
-
```
Comment (alexmojaki) [inline birdseye/__init__.py]: This line should not be removed

File: birdseye/__init__.py [hunk 2]
```diff
@@ -191,6 +191,12 @@ def extract_values(iteration, path):
         session.commit()
 
     def __call__(self, func):
+        if inspect.isclass(func):
+            for name, meth in func.__dict__.items():
```
Comment (alexmojaki) [inline birdseye/__init__.py]: Use `iteritems(func.__dict__)`

File: birdseye/__init__.py [hunk 3]
```diff
@@ -191,6 +191,12 @@ def extract_values(iteration, path):
...
+                if inspect.ismethod(meth) or inspect.isfunction(meth):
+                    setattr(func, name, self.__call__(meth))
```
Comment (alexmojaki) [inline birdseye/__init__.py]: Since `func` isn't really a function this is a bit confusing. Inside the first `if` set `cls = func` and use `cls` instead.

**Comment by alexmojaki** [comment]
Awaiting tests

**Comment by bugosiphionah** [comment]
@alexmojaki  ready for review.

File: tests/test_birdseye.py [hunk 4]
```diff
@@ -275,6 +275,26 @@ def f():
...
+            call_meth = fooz
+            def barz(self):
+                return 'class decorator test'
+
+        def get_stuff(method):
+            return get_call_stuff(get_call_ids(method)[0]).call
+
+        x = Testclass()
+        call = get_stuff(x.barz)
+        self.assertEqual("'class decorator test'", call.return_value)
```
Comment (alexmojaki) [inline tests/test_birdseye.py]: this is still duplicated, put the assert inside the local function and rename the function.

Comment (bugosiphionah) [reply tests/test_birdseye.py]: @alexmojaki  I made the updates.

**Comment by alexmojaki** [comment]
Well done! Try to write more descriptive commit messages though.

**Pull request merged**